# Dublin Aiport Climate Analysis

The data was obtained from Met Éireann's database. We have a csv file with daily data over the period of 01-jan-1942 until 31-jul-2024, and another csv file with monthly averages from the same period.

In [79]:
import pandas as pd
from datetime import datetime

#importing the data into a Pandas data frame with the correct date parsing
dublin_month = pd.read_csv('../Met Eireann Data/dublin_airport_month.csv', skiprows=19)
dublin_daily = pd.read_csv('../Met Eireann Data/dublin_airport_daily.csv', skiprows=25)

# Convert 'date' column to datetime
dublin_daily['date'] = pd.to_datetime(dublin_daily['date'], format='%d-%b-%Y')

# Extract year and month from 'date' column
dublin_daily['year'] = dublin_daily['date'].dt.year
dublin_daily['month'] = dublin_daily['date'].dt.month


We will be using Dash's interactive app library. Install dash to run the apps in the notebook

In [ ]:
pip install dash

## Overview of the data

Monthly data:

Station Name: DUBLIN AIRPORT
Station Height: 71 M 
Latitude:53.428  ,Longitude: -6.241


year:  -  Year
month: -  Month
rain:  -  Precipitation Amount (mm)
meant: -  Mean Air Temperature (C)
maxtp: -  Maximum Air Temperature (C)	  
mintp: -  Minimum  Air Temperature (C)	
mnmax: -  Mean Maximum Temperature (C)
mnmin: -  Mean Minimum Temperature (C)
gmin:  -  Grass Minimum Temperature (C)
wdsp:  -  Mean Wind Speed (knot)
mxgt:  -  Highest Gust (knot)
sun:   -  Sunshine duration (hours)

In [59]:
import dash
import dash_table
import pandas as pd

# Create a Dash app

app = dash.Dash(__name__)

# Define the layout of the app
app.layout = dash_table.DataTable(
    id='table',
    columns=[{'name': i, 'id': i} for i in dublin_month.columns],
    data=dublin_month.to_dict('records'),
    editable=True,
    filter_action='native',
    sort_action='native',
    sort_mode='multi',
    row_selectable='single',
    row_deletable=True,
    selected_rows=[],
    page_action='native',
    page_current=0,
    page_size=10, 
    style_table={'overflowY': 'scroll'}
)

# Run the app
app.run_server(mode='jupyterlab')

In [60]:
dublin_month.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 993 entries, 0 to 992
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   year    993 non-null    int64  
 1   month   993 non-null    int64  
 2   meant   993 non-null    float64
 3   maxtp   993 non-null    float64
 4   mintp   993 non-null    float64
 5   mnmax   993 non-null    float64
 6   mnmin   993 non-null    float64
 7   rain    993 non-null    float64
 8   gmin    993 non-null    object 
 9   wdsp    993 non-null    float64
 10  maxgt   993 non-null    object 
 11  sun     993 non-null    float64
dtypes: float64(8), int64(2), object(2)
memory usage: 93.2+ KB


In [61]:
dublin_month[['maxtp', 'mintp', 'mnmin', 'mnmax', 'rain', 'sun', 'wdsp']].describe().round(2)

maxtp   mintp   mnmin   mnmax    rain     sun    wdsp
count  993.00  993.00  993.00  993.00  993.00  993.00  993.00
mean    17.79    0.81    6.13   13.03   63.14  122.09   10.20
std      4.38    4.07    3.43    4.28   34.38   54.92    2.13
min      4.80  -12.20   -3.50    1.90    3.60   16.40    0.00
25%     13.90   -2.20    3.30    9.40   39.10   73.50    8.70
50%     17.40    0.30    5.50   12.70   56.90  119.00   10.00
75%     21.60    4.20    9.20   17.00   83.20  161.60   11.60
max     29.10   10.00   13.60   22.90  217.00  305.90   17.70

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>maxtp</th>
      <th>mintp</th>
      <th>mnmin</th>
      <th>mnmax</th>
      <th>rain</th>
      <th>sun</th>
      <th>wdsp</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>count</th>
      <td>993.00</td>
      <td>993.00</td>
      <td>993.00</td>
      <td>993.00</td>
      <td>993.00</td>
      <td>993.00</td>
      <td>993.00</td>
    </tr>
    <tr>
      <th>mean</th>
      <td>17.79</td>
      <td>0.81</td>
      <td>6.13</td>
      <td>13.03</td>
      <td>63.14</td>
      <td>122.09</td>
      <td>10.20</td>
    </tr>
    <tr>
      <th>std</th>
      <td>4.38</td>
      <td>4.07</td>
      <td>3.43</td>
      <td>4.28</td>
      <td>34.38</td>
      <td>54.92</td>
      <td>2.13</td>
    </tr>
    <tr>
      <th>min</th>
      <td>4.80</td>
      <td>-12.20</td>
      <td>-3.50</td>
      <td>1.90</td>
      <td>3.60</td>
      <td>16.40</td>
      <td>0.00</td>
    </tr>
    <tr>
      <th>25%</th>
      <td>13.90</td>
      <td>-2.20</td>
      <td>3.30</td>
      <td>9.40</td>
      <td>39.10</td>
      <td>73.50</td>
      <td>8.70</td>
    </tr>
    <tr>
      <th>50%</th>
      <td>17.40</td>
      <td>0.30</td>
      <td>5.50</td>
      <td>12.70</td>
      <td>56.90</td>
      <td>119.00</td>
      <td>10.00</td>
    </tr>
    <tr>
      <th>75%</th>
      <td>21.60</td>
      <td>4.20</td>
      <td>9.20</td>
      <td>17.00</td>
      <td>83.20</td>
      <td>161.60</td>
      <td>11.60</td>
    </tr>
    <tr>
      <th>max</th>
      <td>29.10</td>
      <td>10.00</td>
      <td>13.60</td>
      <td>22.90</td>
      <td>217.00</td>
      <td>305.90</td>
      <td>17.70</td>
    </tr>
  </tbody>
</table>
</div>

Daily data:

Station Name: DUBLIN AIRPORT
Station Height: 71 M 
Latitude:53.428  ,Longitude: -6.241


date:   -  00 to 00 utc
rain:   -  Precipitation Amount (mm)
maxtp:  -  Maximum Air Temperature (C)	  
mintp:  -  Minimum  Air Temperature (C)
gmin:   -  09utc Grass Minimum Temperature (C)
soil:   -  Mean 10cm Soil Temperature (C)
wdsp:   -  Mean Wind Speed (knot)
hm:     -  Highest ten minute mean wind speed (knot)		                 
ddhm:   -  Wind Direction at max 10 min. mean (deg)                  
hg:     -  Highest Gust (knot)
cbl:    -  Mean CBL Pressure (hpa)
sun:    -  Sunshine duration (hours)
g_rad:  -  Global Radiation (j/cm sq.)
pe:     -  Potential Evapotranspiration (mm)				
evap:   -  Evaporation (mm)						
smd_wd: -  Soil Moisture Deficits(mm) well drained		
smd_md: -  Soil Moisture Deficits(mm) moderately drained	
smd_pd: -  Soil Moisture Deficits(mm) poorly drained 

We can drop a few columns in the daily data table since we don't need to look at all the categories.

In [80]:
dublin_daily = dublin_daily.drop(['gmin','ind','ind.1','ind.2','ind.3','ind.4','ind.5','igmin', 'dos', 'soil', 'hm','hg', 'ddhm', 'cbl', 'g_rad', 'pe', 'evap', 'smd_wd', 'smd_md', 'smd_pd'], axis=1)

In [81]:
dublin_daily

date  maxtp  mintp  rain  wdsp   sun  year  month
0     1942-01-01    9.7    6.8   0.0  17.2   0.0  1942      1
1     1942-01-02    9.9    7.9   0.1  15.2   0.0  1942      1
2     1942-01-03   11.2    8.9   1.5  14.0   0.1  1942      1
3     1942-01-04    9.2    2.7   3.5  17.0   0.6  1942      1
4     1942-01-05    3.5   -0.8   0.6  13.0   3.4  1942      1
...          ...    ...    ...   ...   ...   ...   ...    ...
30158 2024-07-27   20.3   12.3   0.3   5.2   3.3  2024      7
30159 2024-07-28   21.9   10.9   0.0   6.5   9.8  2024      7
30160 2024-07-29   22.5   12.8   1.2   6.0   7.1  2024      7
30161 2024-07-30   20.6    9.9   0.0   5.9  11.1  2024      7
30162 2024-07-31   20.9    8.1   0.0   4.9  11.2  2024      7

[30163 rows x 8 columns]

In [82]:
import dash
import dash_table

# Create a Dash app

app = dash.Dash(__name__)

# Define the layout of the app
app.layout = dash_table.DataTable(
    id='table',
    columns=[{'name': i, 'id': i} for i in dublin_daily.columns],
    data=dublin_daily.to_dict('records'),
    editable=True,
    filter_action='native',
    sort_action='native',
    sort_mode='multi',
    row_selectable='single',
    row_deletable=True,
    selected_rows=[],
    page_action='native',
    page_current=0,
    page_size=20, 
    style_table={'overflowY': 'scroll'}
)

# Run the app
app.run_server(mode='jupyterlab')

In [83]:
dublin_daily.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30163 entries, 0 to 30162
Data columns (total 8 columns):
 #   Column  Non-Null Count  Dtype         
---  ------  --------------  -----         
 0   date    30163 non-null  datetime64[ns]
 1   maxtp   30163 non-null  float64       
 2   mintp   30163 non-null  float64       
 3   rain    30163 non-null  float64       
 4   wdsp    30163 non-null  float64       
 5   sun     30163 non-null  float64       
 6   year    30163 non-null  int32         
 7   month   30163 non-null  int32         
dtypes: datetime64[ns](1), float64(5), int32(2)
memory usage: 1.6 MB


In [87]:
dublin_daily[['maxtp', 'mintp', 'rain', 'wdsp', 'sun']].describe().round(2)

maxtp     mintp      rain      wdsp       sun
count  30163.00  30163.00  30163.00  30163.00  30163.00
mean      13.06      6.15      2.07     10.19      4.02
std        4.91      4.38      4.39      4.60      3.76
min       -4.70    -12.20      0.00      0.00      0.00
25%        9.40      2.90      0.00      6.80      0.50
50%       13.00      6.30      0.20      9.50      3.20
75%       16.90      9.60      2.20     13.00      6.50
max       29.10     18.40     92.60     35.50     15.90

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>maxtp</th>
      <th>mintp</th>
      <th>rain</th>
      <th>wdsp</th>
      <th>sun</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>count</th>
      <td>30163.00</td>
      <td>30163.00</td>
      <td>30163.00</td>
      <td>30163.00</td>
      <td>30163.00</td>
    </tr>
    <tr>
      <th>mean</th>
      <td>13.06</td>
      <td>6.15</td>
      <td>2.07</td>
      <td>10.19</td>
      <td>4.02</td>
    </tr>
    <tr>
      <th>std</th>
      <td>4.91</td>
      <td>4.38</td>
      <td>4.39</td>
      <td>4.60</td>
      <td>3.76</td>
    </tr>
    <tr>
      <th>min</th>
      <td>-4.70</td>
      <td>-12.20</td>
      <td>0.00</td>
      <td>0.00</td>
      <td>0.00</td>
    </tr>
    <tr>
      <th>25%</th>
      <td>9.40</td>
      <td>2.90</td>
      <td>0.00</td>
      <td>6.80</td>
      <td>0.50</td>
    </tr>
    <tr>
      <th>50%</th>
      <td>13.00</td>
      <td>6.30</td>
      <td>0.20</td>
      <td>9.50</td>
      <td>3.20</td>
    </tr>
    <tr>
      <th>75%</th>
      <td>16.90</td>
      <td>9.60</td>
      <td>2.20</td>
      <td>13.00</td>
      <td>6.50</td>
    </tr>
    <tr>
      <th>max</th>
      <td>29.10</td>
      <td>18.40</td>
      <td>92.60</td>
      <td>35.50</td>
      <td>15.90</td>
    </tr>
  </tbody>
</table>
</div>

# Visualisations

## Temperatures